libraries/AC_AutoTune/AC_AutoTune.cpp

```diff
@@ -270,20 +270,16 @@ void AC_AutoTune::run()
 // return true if vehicle is close to level
 bool AC_AutoTune::currently_level()
 {
-    float threshold_mul = 1.0;
-
-    uint32_t now_ms = AP_HAL::millis();
-    if (now_ms - level_start_time_ms > AUTOTUNE_LEVEL_TIMEOUT_MS) {
-        // after a long wait we use looser threshold, to allow tuning
-        // with poor initial gains
-        threshold_mul *= 2;
-    }
+    const uint32_t now_ms = AP_HAL::millis();
 
-    // display warning if vehicle fails to level
-    if ((now_ms - level_start_time_ms > AUTOTUNE_LEVEL_WARNING_INTERVAL_MS) &&
-        (now_ms - level_fail_warning_time_ms > AUTOTUNE_LEVEL_WARNING_INTERVAL_MS)) {
-        gcs().send_text(MAV_SEVERITY_CRITICAL, "AutoTune: failing to level, please tune manually");
-        level_fail_warning_time_ms = now_ms;
+    // slew threshold to ensure settling time
+    // relax threshold if we pass AUTOTUNE_LEVEL_TIMEOUT_MS
+    float threshold_mul = constrain_float((float)(now_ms - level_start_time_ms) / (float)AUTOTUNE_LEVEL_TIMEOUT_MS, 0.0, 2.0);
```

Comment:
```suggestion
    const float threshold_mul = constrain_float((float)(now_ms - level_start_time_ms) / (float)AUTOTUNE_LEVEL_TIMEOUT_MS, 0.0, 2.0);
```

... and move this calculation to after the `AUTOTUNE_LEVEL_TIMEOUT_MS` check

This is a variable-scope-minimisation thing